resdk/data_upload/annotate_samples.py

```diff
@@ -0,0 +1,85 @@
+"""Annotate samples in a collection."""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+from resdk.data_upload.samplesheet import FileImporter, Sample
+
+def annotate_samples(collection, samplesheet_name, local_sheet=True,
+                     verbose=True):
+    """Annotate the samples in a collection based on a sample annotation
+    spreadsheet.
+    """
+
+    # Extract the information from a sample annotation spreadsheet
+    # Validate each sample annotation
+    if local_sheet:
+        samplesheet = FileImporter(samplesheet_name)
+        err_samples = []
+        missing_samples = []
+        sample_count = 0
+
+    # TODO: Try to pull the samplesheet as a data object from the collection
+    #       once it is possible to upload samplesheets
+    # else:
+    #     sheet = collection.data.get(name=samplesheet_name,
+    #                                 type='data:annotation')
+    #     sheet.download()
+    #     samplesheet = FileImporter(sheet.files()[0])
+    #     err_samples = []
+
+    # Apply each annotation to a sample in the collection
+    for entry in samplesheet.sample_list:
+        try:
+            # Find the existing (unannotated) samples
+            new_samples = collection.samples.filter(name=entry['SAMPLE_NAME'])
+            if not new_samples:
+                raise NameError(
+                    "Sample `{}` not found.".format(entry['SAMPLE_NAME'])
+                )
+
+            # Create and validate the sample descriptor
+            sample_info = Sample(entry)
+
+            # Apply the sample descriptor to the sample(s)
+            for s in new_samples: # pylint: disable=C0103
+                s.descriptor_schema = 'sample'
```

Comment:
Is it ok to always use the same descriptor? We have many of them: https://github.com/genialis/resolwe-bio/tree/master/resolwe_bio/descriptors